src/native/util/os_linux.cpp

```diff
@@ -27,25 +27,24 @@ get_current_uid()
 const uid_t ThreadScope::orig_uid = getuid();
 const gid_t ThreadScope::orig_gid = getgid();
 
-const std::vector<gid_t> ThreadScope::orig_groups = [] {
-    std::vector<gid_t> groups(NGROUPS_MAX);
-    int r = getgroups(NGROUPS_MAX, &groups[0]);
-    groups.resize(r);
-    return groups;
-}();
+const std::vector<gid_t> ThreadScope::orig_groups = get_process_groups();
 
 /**
- * set the effective uid/gid of the current thread using direct syscalls
- * unsets the supplementary group IDs for the current thread using direct syscall
+ * set the effective uid/gid/supplemental_groups of the current thread using direct syscalls
  * we have to bypass the libc wrappers because posix requires it to syncronize
- * uid & gid to all threads which is undesirable in our case.
+ * uid, gid & supplemental_groups to all threads which is undesirable in our case.
```

Comment:
Do you know if this comment is still relevant?
If so, could you explain it?

Reply:
yes it is. I assume the libc functions change the uid, gid, supplemental groups for all threads. we just want to change it for one thread. so we use special functions. we need to validate it works on mac as well